--- apertium-ca-it-0.1.0/Makefile.am.orig	2011-01-13 12:27:13.000000000 +0100
+++ apertium-ca-it-0.1.0/Makefile.am	2011-01-30 20:16:15.652686361 +0100
@@ -106,15 +106,15 @@
                     $(PREFIX2).autopgen.bin \
 		    $(PREFIX1).t1x.bin $(PREFIX1).t2x.bin $(PREFIX1).t3x.bin \
 		    $(PREFIX2).t1x.bin $(PREFIX2).t2x.bin $(PREFIX2).t3x.bin \
-		    $(PREFIX1).mode $(PREFIX2).mode \
 		    $(PREFIX1).prob $(PREFIX2).prob 
 		  
 install-data-local: 
-	$(INSTALL_DATA) $(PREFIX2).mode $(apertium_ca_modesdir)
-	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t1x $(BASENAME).$(PREFIX1).t2x $(apertium_ca_itdir)
-	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t3x $(BASENAME).$(PREFIX2).t1x $(apertium_ca_itdir)
-	$(INSTALL_DATA) $(BASENAME).$(PREFIX2).t2x $(BASENAME).$(PREFIX2).t3x $(apertium_ca_itdir)
+	apertium-gen-modes modes.xml apertium-$(PREFIX1)
+	$(INSTALL_DATA) $(PREFIX2).mode $(DESTDIR)$(apertium_ca_modesdir)
+	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t1x $(BASENAME).$(PREFIX1).t2x $(DESTDIR)$(apertium_ca_itdir)
+	$(INSTALL_DATA) $(BASENAME).$(PREFIX1).t3x $(BASENAME).$(PREFIX2).t1x $(DESTDIR)$(apertium_ca_itdir)
+	$(INSTALL_DATA) $(BASENAME).$(PREFIX2).t2x $(BASENAME).$(PREFIX2).t3x $(DESTDIR)$(apertium_ca_itdir)
 
-#	$(INSTALL_DATA) $(PREFIX1).mode $(apertium_ca_modesdir)
+#	$(INSTALL_DATA) $(PREFIX1).mode $(DESTDIR)$(apertium_ca_modesdir)
 
 CLEANFILES = -rf $(TARGETS_COMMON) modes
